File: micro-auth-oidc/src/main/java/com/jiangtj/micro/auth/oidc/JwtProperties.java
package com.jiangtj.micro.auth.oidc;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.ArrayList;
import java.util.List;

@Data
@ConfigurationProperties(prefix = "jmicro.jwt")
public class JwtProperties {
    private List<OidcProperties> oidc = new ArrayList<>();
}

File: micro-auth-oidc/src/main/java/com/jiangtj/micro/auth/oidc/JwtProperties.kt
package com.jiangtj.micro.auth.oidc

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "micro.jwt")
data class JwtProperties(
    var oidc: List<OidcProperties> = emptyList(),
)

enum class MatcherStyle {
    PREFIX,
    REGEX,
    ANT
}
